crates/node/src/main.rs

```diff
@@ -108,6 +109,18 @@ async fn main() -> anyhow::Result<()> {
         l1_gas_setter.update_strk_l1_data_gas_price(strk_fix_blob_gas as u128);
         l1_gas_setter.set_strk_data_gas_price_sync_enabled(false);
     }
+    if let Some(ref oracle_url) = run_cmd.l1_sync_params.oracle_url {
+        if let Some(ref oracle_api_key) = run_cmd.l1_sync_params.oracle_api_key {
+            let oracle = Oracle::new("Pragma", oracle_url.to_string(), oracle_api_key.clone())?;
+            l1_gas_setter.set_oracle_provider(oracle);
+        }
+    }
+
+    if l1_gas_setter.is_oracle_needed() && l1_gas_setter.oracle_provider.is_none() {
+        log::error!("STRK gas is not fixed and oracle is not provided");
+        panic!();
```

Comment:
I would personally handle the case where we dont want the strk price by defaulting it

Reply:
also, could you use anyhow::bail! instead of panic and put the error message inside the bail?